docs/guides/auth/oauth.rst

```diff
@@ -1,15 +1,225 @@
+        const redirectUrl = new URL("authorize", EDGEDB_AUTH_BASE_URL);
+        redirectUrl.searchParams.set("provider", provider);
+        redirectUrl.searchParams.set("challenge", pkce.challenge);
+        redirectUrl.searchParams.set(
+          "redirect_to",
+          `http://localhost:${SERVER_PORT}/auth/callback`,
+        );
+
+        res.writeHead(302, {
+          "Set-Cookie": `edgedb-pkce-verifier=${pkce.verifier}; HttpOnly; Path=/; Secure; SameSite=Strict`,
+          Location: redirectUrl.href,
+        });
+        res.end();
+      };
+
+   .. lint-on
+
+3. At the very end of the flow, the EdgeDB server will redirect the user's
+   browser to the ``redirect_to`` address with a single query parameter:
+   ``code``. This route should be a server route that has access to the
+   ``verifier``. You then take that ``code`` and the ``verifier`` you stored in
+   step 1, and make a request to the EdgeDB Auth extension to exchange these
+   two pieces of data for an ``auth_token``.
+
+   .. lint-off
+
+   .. code-block:: javascript
+
+      /**
+       * Handles the PKCE callback and exchanges the `code` and `verifier
+       * for an auth_token, setting the auth_token as an HttpOnly cookie.
+       *
+       * @param {Request} req
+       * @param {Response} res
+       */
+      const handleCallback = async (req, res) => {
+         const requestUrl = getRequestUrl(req);
+
+         const code = requestUrl.searchParams.get("code");
+         if (!code) {
+            const error = requestUrl.searchParams.get("error");
+            res.status = 400;
+            res.end(
+               `OAuth callback is missing 'code'. \
+      OAuth provider responded with error: ${error}`,
+            );
+            return;
+         }
+
+         const cookies = req.headers.cookie?.split("; ");
+         const verifier = cookies
+            ?.find((cookie) => cookie.startsWith("edgedb-pkce-verifier="))
+            ?.split("=")[1];
+         if (!verifier) {
+            res.status = 400;
+            res.end(
+               `Could not find 'verifier' in the cookie store. Is this the \
+      same user agent/browser that started the authorization flow?`,
+            );
+            return;
+         }
+
+         const codeExchangeUrl = new URL("token", EDGEDB_AUTH_BASE_URL);
+         codeExchangeUrl.searchParams.set("code", code);
+         codeExchangeUrl.searchParams.set("verifier", verifier);
+         const codeExchangeResponse = await fetch(codeExchangeUrl.href, {
+            method: "GET",
+         });
+
+         if (!codeExchangeResponse.ok) {
+            const text = await codeExchangeResponse.text();
+            res.status = 400;
+            res.end(`Error from the auth server: ${text}`);
+            return;
+         }
+
+         const { auth_token } = await codeExchangeResponse.json();
+         res.writeHead(204, {
+            "Set-Cookie": `edgedb-auth-token=${auth_token}; HttpOnly; Path=/; Secure; SameSite=Strict`,
```

Comment:
This line is too long.

Reply:
Yeah, I mean I can _force_ this line to be the right length, but I think readability and understandability suffer by like breaking it up with a `[].join("; ")` or using escapes to remove new lines or something like that. We're also trying to avoid too many helper functions, but that's another option.